gobblin-modules/gobblin-azkaban/src/main/java/org/apache/gobblin/service/modules/orchestration/AzkabanSpecProducer.java

```diff
@@ -84,6 +83,18 @@ public void close() throws IOException {
   public Future<?> addSpec(Spec addedSpec) {
     // If project already exists, execute it
     try {
+      // If encountered Azkaban authentication issue on start up, attempt authentication again.
```

Comment:
What happens if we didn't modify this code at all? I thought it should auto attempt the authentication again because the sessionId is null, and the sessionCreationTime is `-1` which will fulfill the renew requirements before making a API request.